✨ 오피스 컬렉션뷰와 오피스 컬렉션뷰셀을 구성했습니다.

BrightHyeon opened the pull request:
# 배경
- 홈 화면 상단에 각종 오피스들이 나열되어있는 가로 스크롤 컬렉션뷰가 필요합니다.

# 작업 내용
- 각종 Cell type, Header, Footer 등이 상속하는 UICollectionReuableView를 extension해서 자신의 클레스 네임을 고유한 식별자로 사용하도록 했습니다.
- UICollectionView를 extension하여 Cell register 과정을 간소화시켰습니다.
- 마찬가지로 extension하여 dequeueReuableCell을 등록하는 과정을 간소화시켰습니다. (명시적인 선에서)
- 가로 스크롤 컬렉션뷰의 재사용성을 위해 HorizontalCollectionView를 만들었습니다. (UICollectionView를 확장하려다가 그렇게하면 의미전달이 모호하고, vertical case까지 고려해서 추가작업을 해야하기에 아예 따로 만듦)
- 기존 폰트들을 Design Guide에서 제공된 font로 변경하였습니다.
- 중복되는 import 제거하였습니다.

# 추가 작업 내용
- stackView에 속하는 컴포넌트들의 translates... = false 코드 제거. (오토레이아웃 방식이 아니라 불필요)
- stackView를 선언할 때 addArrangedSubview를 하는 것으로 수정. (속한 컴포넌트 명확. addSubview 로직과의 명확한 구분)

# 테스트 방법
- 현재 Data Manager 로직이 완성되지않아서 UI 확인용 가짜 데이터를 넣은 상황입니다.
- 시뮬레이션을 켜서 바로 눈으로 확인 가능합니다.

# 리뷰 노트
- https://nemecek.be/blog/82/useful-extensions-for-collectionview-and-compositional-layout

# 스크린샷
<img width="150" alt="image" src="https://user-images.githubusercontent.com/95853235/196959216-066e47b5-7538-4063-92a4-bf1a90ff8ad3.png">

BrightHyeon committed:
✨ 자신의 고유한 class name을 식별자로써 사용하도록 UICollectionReuableView 확장

BrightHyeon committed:
✨ 복잡한 register과정과 dequeueReusable과정을 간소화하기 위한 UICollectionView extension

BrightHyeon committed:
✨ 가로스크롤 용 CollectionView 선언

BrightHyeon committed:
✨ 오피스컬렉션뷰 속성 설정 및 레이아웃 추가

BrightHyeon committed:
✨ 오피스컬렉션뷰 셀을 구성하는 컴포넌트들 정의

BrightHyeon committed:
✨ 오피스컬렉션뷰 셀 서브뷰들의 레이아웃 설정

BrightHyeon committed:
✨ isActive 방식에서 NSLayoutConstraints.activate() 방식으로 변경

BrightHyeon committed:
✨ 오피스 모델 생성

BrightHyeon committed:
✨ officeCollectionView 구성 완료

BrightHyeon committed:
✨ 이미지 위에 옅은 검정 layer 추가

BrightHyeon committed:
✏️ 중복되는 import 제거

hs-ryu reviewed (commented):
[P5] HomeView 파일 이름을 HomeViewController로 변경하는 것은 별로일까요?

DevLuce commented on Workade/Views&ViewModels/Home/OfficeCollectionViewCell.swift:
```diff
@@ -0,0 +1,121 @@
+        let label = UILabel()
+        label.font = .customFont(for: .subHeadline)
+        label.textColor = .white
+        label.translatesAutoresizingMaskIntoConstraints = false
```
[P4] 스택뷰 내에 위치하는 뷰에 경우에는 이 코드가 불필요한 것 같습니다!

BrightHyeon replied:
translatesAutoresizingMaskIntoContraints = false 코드만 해당하는 것인가요? 아니면 label에 대한 설정이 불필요한 걸까요??

DevLuce replied:
앗 translatesAutoresizingMaskIntoContraints = false 만이요!!
저거만 체크했는데.. 4줄이 됐네요

BrightHyeon replied:
오 꿀팁 감사합니다! stackView안에서의 정렬은 오토레이아웃이 아니군요! 굳굳~ 수정하겠습니다

DevLuce commented on Workade/Views&ViewModels/Home/OfficeCollectionViewCell.swift:
```diff
@@ -0,0 +1,121 @@
+        let label = UILabel()
+        label.font = .customFont(for: .title3)
+        label.textColor = .white
+        label.translatesAutoresizingMaskIntoConstraints = false
```
[P4] 이 코드도 마찬가지로 스택뷰 내에 위치하여 불필요한 것 같습니다.

DevLuce commented on Workade/Views&ViewModels/Home/OfficeCollectionViewCell.swift:
```diff
@@ -0,0 +1,121 @@
+        contentView.addSubview(mapButton)
+        contentView.addSubview(titleStackView)
+        titleStackView.addArrangedSubview(regionNameLabel)
+        titleStackView.addArrangedSubview(officeNameLabel)
```
[P5] 저는 스택뷰를 정의할 때에 이 코드를 작성하였는데 컨벤션 다시 한번 얘기해보면 좋을것 같아요!

BrightHyeon replied:
이 방법은 좋은 것 같습니다! lazy var로 선언하여 stackView 클로저안에서 작성하는 방법을 말씀하신 것이 맞나요??

DevLuce replied:
네 맞습니다!! 스택뷰 클로저 안에서요!!

BrightHyeon replied:
굳굳 수정하겠습니당~!

BrightHyeon committed:
♻️ 불필요한 설정 제거 (스택뷰 소속 컴포넌트의 translates)

BrightHyeon committed:
♻️ stackView를 선언할 때 addArrangedSubview를 하는 것으로 수정

해당 stackView에 속한 컴포넌트를 보다 명확하게 알 수 있음
addSubview로직과 addArrangedSuview 로직을 명확히 구분할 수 있음

DevLuce reviewed (commented):
전 확인 완료했습니다!

E-know commented:
[P1] Merge로 인해서 컨플릭트 해결 부탁드립니다!

BrightHyeon committed:
Merge branch 'main' into feat/OfficeCollectionView

BrightHyeon commented:
> [P1] Merge로 인해서 컨플릭트 해결 부탁드립니다!

> Merge Conflict 해소했습니다!

E-know reviewed (approved):
확인 했습니다!
